src/Helper/OrderHelper.php

```diff
@@ -482,38 +495,103 @@ public function extractName($name)
 		);
 	}
 
-	/**
-	 * Check if order was already imported.
-	 *
-	 * @param mixed $externalOrderId
-	 *
-	 * @return bool
-	 */
-	public function orderWasImported($externalOrderId)
-	{
-		/** @var OrderRepositoryContract $orderRepo */
-		$orderRepo = pluginApp(OrderRepositoryContract::class);
-
-		if($orderRepo instanceof OrderRepositoryContract)
-		{
-			$orderRepo->setFilters([
-				                       'externalOrderId' => $externalOrderId,
-				                       'referrerId'      => $this->getReferrerId(),
-				                       'orderType'       => OrderType::TYPE_SALES_ORDER,
-			                       ]);
-
-			/** @var PaginatedResult $paginatedResult */
-			$paginatedResult = $orderRepo->searchOrders();
-
-			if($paginatedResult instanceof PaginatedResult)
-			{
-				if($paginatedResult->getTotalCount() > 0)
-				{
-					return true;
-				}
-			}
-		}
-
-		return false;
-	}
+    /**
+     * Create payment.
+     *
+     * @param array $data
+     * @param Order $order
+     */
+    public function createPayment(array $data, Order $order)
+    {
+        try {
+            /** @var PaymentService $paymentService */
+            $paymentService = pluginApp(PaymentService::class);
+
+            /** @var PaymentRepositoryContract $paymentRepo */
+            $paymentRepo = pluginApp(PaymentRepositoryContract::class);
+
+            $payments = $paymentService->findShopPaymentByReceipt($this->settingsHelper->getShopSettings('shopId'), $data['receipt_id']);
+
+            if (is_array($payments) && count($payments)) {
+                /** @var PaymentHelper $paymentHelper */
+                $paymentHelper = $this->app->make(PaymentHelper::class);
+
+                foreach ($payments as $paymentData) {
+                    $createPaymentTimeBerlin = Carbon::createFromTimestamp($paymentData['create_date'])
+                        ->timezone('Europe/Berlin')
+                        ->format('Y-m-d H:i:s');
+
+                    $updatePaymentTimeBerlin = Carbon::createFromTimestamp($paymentData['update_date'])
+                        ->timezone('Europe/Berlin')
+                        ->format('Y-m-d H:i:s');
+
+                    /** @var Payment $payment */
+                    $payment                   = $this->app->make(Payment::class);
+                    $payment->amount           = ($paymentData['amount_gross'] / 100) - $data['total_tax_cost'];
+                    $payment->mopId            = $paymentHelper->getPaymentMethodId();
```

Comment:
How does the payment helper know which ID he has to take? Isnt there a Etsy config thing where the ID is stored?